src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -347,67 +331,33 @@ object NamespaceWithWorkflow {
    for {
       call <- workflow.calls
       (name, expression) <- call.inputMappings
-      memberAccess <- expression.ast.findTopLevelMemberAccesses()
-    } yield {
-      getCallFromMemberAccessAst(memberAccess, workflow, wdlSyntaxErrorFormatter).get
-    }
+      memberAccessAst <- expression.ast.findTopLevelMemberAccesses()
+    } validateMemberAccessAst(memberAccessAst, workflow, wdlSyntaxErrorFormatter)
 
     new NamespaceWithWorkflow(namespace, workflow, imports, namespaces, tasks, terminalMap, wdlSyntaxErrorFormatter, ast)
   }
 
-  // FIXME/TODO: Depending on how things work w/ the related FIXME's in the actual case class this might change. This is also being used as a validity check. I had another FIXME about this up above
-  /*
-   * Partially evaluate MemberAccess ASTs to make sure they're not nonsense at compile time
+  /**
+   * Ensures that the lhs corresponds to a call and the rhs corresponds to one if its outputs.
    *
-   * MemberAccess ASTs are of the form lhs.rhs
+   * Run for its side effect (i.e. Exception) but we were previously using a Try and immediately calling .get on it
+   * so its the same thing
    */
-  def getCallFromMemberAccessAst(ast: Ast, workflow: Workflow, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Try[Call] = {
-    def callFromName(name: String): Try[Call] = {
-      workflow.calls.find(_.name == name) match {
-        case Some(c: Call) => Success(c)
-        case _ => Failure(new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast)))
-      }
-    }
-    val rhs = ast.getAttribute("rhs").sourceString
-
-    /**
-     * The right-hand side of a member-access AST should always be interpreted as a String
-     * Sometimes, the left-hand side is itself a MemberAccess AST, like in the expression
-     * for `call t1` below.  In that example, callFromName("ns.ns2.task_name") would be
-     * called.  In the `call t2` example, callFromName("alias") is called
-     *
-     * import "test.wdl" as ns
-     * workflow w {
-     *  call ns.ns2.task_name
-     *  call t1 {
-     *    input: x=ns.ns2.task_name.output
-     *  }
-     *
-     *  call ns.ns2.task_name as alias
-     *  call t2 {
-     *    input: y=alias.output
-     *  }
-     *}
-     */
-    val lhs = callFromName(ast.getAttribute("lhs") match {
-      case a: Ast => WdlExpression.toString(a)
-      case terminal: Terminal => terminal.sourceString
-    })
-
-    lhs match {
-      case Success(c: Call) =>
-        c.task.outputs.find {_.name == rhs}.getOrElse {
-          throw new SyntaxError(wdlSyntaxErrorFormatter.memberAccessReferencesBadTaskInput(ast))
-        }
-        Success(c)
-      case f => f
+  def validateMemberAccessAst(memberAccessAst: Ast,
+                              workflow: Workflow,
+                              errorFormatter: WdlSyntaxErrorFormatter): Unit = {
+    val memberAccess = MemberAccess(memberAccessAst)
+    val call = workflow.calls find { _.name == memberAccess.lhs }
```

Comment:
See my comment in one of the larger threads above re MemberAccess and FQN vs lhs. It looks like the original validation was only looking at name, not FQN so perhaps that's the right thing to do here anyways.